core/Pos/Binary/Core/Blockchain.hs

```diff
@@ -35,13 +34,16 @@ instance ( Typeable b
     decode = do
         enforceSize "GenericBlockHeader b" 5
         blockMagic <- decode
+        -- TODO include ProtocolMagic in the definition of GenericBlockHeader,
+        -- and decode it, eliminating this failure case. Protocol magic checks
+        -- must not happen in decoding.
         when (blockMagic /= getProtocolMagic protocolMagic) $ cborError $
-            "GenericBlockHeader failed with wrong magic: " <> pretty blockMagic
-        prevBlock <- decode
-        bodyProof <- decode
-        consensus <- decode
-        extra     <- decode
-        toCborError $ T.recreateGenericHeader prevBlock bodyProof consensus extra
+            "GenericBlockHeader failed with wrong magic: " <> show blockMagic
```

Comment:
It's certainly minor and negligible, but why did you change `pretty` to `show`? `pretty` should be a bit faster, because it doesn't use `String`.

Reply:
Probably a merge conflict resolution error. Will fix.